front/app/containers/Admin/formBuilder/components/FormBuilderSettings/utils.tsx

```diff
@@ -1,33 +1,35 @@
 import React from 'react';
-import { ICustomFieldInputType } from 'services/formCustomFields';
+import { IFlatCustomFieldWithIndex } from 'services/formCustomFields';
 import { Locale } from 'typings';
 
 // Components
 import ConfigMultiselectWithLocaleSwitcher from './ConfigMultiselectWithLocaleSwitcher';
 import LinearScaleSettings from './LinearScaleSettings';
+import PageSettings from './PageSettings';
 
 // Function to return additional settings based on input type
 export function getAdditionalSettings(
-  type: ICustomFieldInputType,
-  locales: Locale[],
-  index: number
+  field: IFlatCustomFieldWithIndex,
+  locales: Locale[]
 ) {
-  switch (type) {
+  switch (field.input_type) {
     case 'multiselect':
     case 'select':
       return (
         <ConfigMultiselectWithLocaleSwitcher
-          nameInputType={`customFields.${index}.input_type`}
-          name={`customFields.${index}.options`}
+          nameInputType={`customFields.${field.index}.input_type`}
+          name={`customFields.${field.index}.options`}
           locales={locales}
         />
       );
+    case 'page':
+      return PageSettings(field);
```

Comment:
Minor: We could do this here instead and use camelCase for normal functions for uniformity. We would then change the Page Settings where it is defined to 

```
type Props = {
  field: IFlatCustomFieldWithIndex
}

const PageSettings = ({ field }: Props) => (

```


```
return <PageSettings field={field} />;
```

Reply:
For some reason the form context didn't work correctly when I did that - I'll try it again though :)